### PR TITLE
feat: implement who can comment redesign

### PR DESCRIPTION
**Pull request by nitegeist**
## What does this PR do?

Implements the redesign for the `who can comment` section when creating a new post.

## Related issues

Fixes issue #809 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Explanation of the changes

The changes to be implemented were: 
image
image
image

The changes implemented are: 

https://user-images.githubusercontent.com/31413091/230703286-f0d88869-9394-4779-82d4-560ed93a3749.mp4

### CODE REVIEW

**Comment by bigint** [comment]
Wow amazing work @nitegeist 👏🏼 

I'll review the code and design ASAP today

File: apps/web/src/components/Shared/PopoverTransition.tsx
Comment (bigint) [inline apps/web/src/components/Shared/PopoverTransition.tsx:1]: We already have existing component called `MenuTransition` you can remove this

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 1]
```diff
@@ -547,7 +548,7 @@ const NewPublication: FC<NewPublicationProps> = ({ publication }) => {
           <Attachment />
           <Giphy setGifAttachment={(gif: IGif) => setGifAttachment(gif)} />
           <CollectSettings />
-          <ReferenceSettings />
+          <NewReferenceSettings />
```
Comment (bigint) [inline apps/web/src/components/Composer/NewPublication.tsx:551]: Replace old `ReferenceSettings` to new one, don't want to give a new name called `NewReferenceSettings`

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 2]
```diff
@@ -64,8 +64,9 @@ import { PUBLICATION } from 'src/tracking';
 import type { LensterAttachment } from 'src/types';
 import { Button, Card, ErrorMessage, Spinner } from 'ui';
 import { v4 as uuid } from 'uuid';
-import { useContractWrite, useProvider, useSigner, useSignTypedData } from 'wagmi';
+import { useContractWrite, useProvider, useSignTypedData, useSigner } from 'wagmi';
```
Comment (bigint) [inline apps/web/src/components/Composer/NewPublication.tsx:67]: Is this changed by eslint?

Comment (bigint) [reply apps/web/src/components/Composer/NewPublication.tsx:67]: CI and Vercel fails

image

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Shared/PopoverTransition.tsx
Comment (nitegeist) [reply apps/web/src/components/Shared/PopoverTransition.tsx:1]: Got it, I'll remove this and use `MenuTransition` instead

File: apps/web/src/components/Composer/NewPublication.tsx (hunk 1 above)
Comment (nitegeist) [reply apps/web/src/components/Composer/NewPublication.tsx:551]: Will do

File: apps/web/src/components/Composer/NewPublication.tsx (hunk 2 above)
Comment (nitegeist) [reply apps/web/src/components/Composer/NewPublication.tsx:67]: > Is this changed by eslint?

Yeah this was changed by eslint. I'll see if I can revert whatever changes it made

Comment (nitegeist) [reply apps/web/src/components/Composer/NewPublication.tsx:67]: > CI and Vercel fails
> 
> image

:( Perhaps the eslint changes caused it to fail?

**Comment by bigint** [comment]
Still deployment fails 🙇🏼

**Comment by nitegeist** [comment]
> Still deployment fails 🙇🏼

Does it fail with the same errors? 
image

**Comment by bigint** [comment]
Screenshot 2023-04-08 at 12 20 37 PM

**Comment by nitegeist** [comment]
@bigint I've fixed the deploy issue, feel free to check this out when you're able to.

**Comment by bigint** [comment]
We can reduce lot of space, it looks very huge now

image

**Comment by bigint** [comment]
Line spacing of the words looks too big

image

**Comment by bigint** [comment]
Mobile view breaks, I would suggest to turn into modal instead of dropdown for mobile web

image

**Comment by nitegeist** [comment]
> We can reduce lot of space, it looks very huge now
> 
> image



> Line spacing of the words looks too big
> 
> image



> Mobile view breaks, I would suggest to turn into modal instead of dropdown for mobile web
> 
> image

All 3 of these have been addressed in my latest push. I turned used a modal instead of a popover which should solve the mobile issue, and reduced spacing overall as well as line height for the text.

**Comment by nitegeist** [comment]
> > Mobile view breaks, I would suggest to turn into modal instead of dropdown for mobile web
> 
> Modal only for mobile view 🙇🏼 

Oh ok, gotcha. I'll make this change.

**Comment by nitegeist** [comment]
> > Mobile view breaks, I would suggest to turn into modal instead of dropdown for mobile web
> 
> Modal only for mobile view 🙇🏼

This change is now implemented in my latest commit

**Comment by bigint** [comment]
https://user-images.githubusercontent.com/69431456/231664881-d7adb15b-176c-4900-b3f4-6dbd500e7a5e.mov

@nitegeist I can't drag more than one item 🙇🏼

**Comment by nitegeist** [comment]
> @nitegeist I can't drag more than one item 🙇🏼

@bigint I've removed what I think is causing you to be unable to drag more than one item. Unfortunately, I can't test my changes properly anymore since the locally served project requires me to use a lens protocol handle and I don't have one.

**Comment by saagu335** [comment]
@nitegeist you can use testnet account for development

**Comment by nitegeist** [comment]
> @nitegeist you can use testnet account for development

Oh? Thanks for letting me know. I'll give that a try today. Is it just a matter of using the testnet link in the `.env` file?

**Comment by saagu335** [comment]
Yes you can just change `NEXT_PUBLIC_LENS_NETWORK` to testnet

**Comment by nitegeist** [comment]
> @nitegeist I can't drag more than one item 🙇🏼

@bigint I've made the range slider transition more smooth and have it snap to the closest value. I don't think I can get have it use the values **and** have it drag through more than one item at the same time despite my efforts. The range slider doesn't seem to work like that if I'm tying it to specific values in order to trigger an update :(

**Comment by nitegeist** [comment]
> Screen.Recording.2023-04-18.at.8.03.50.PM.mov 
> Still, this is acting weird 😕 I can't drag beyond one item in a single drag

@bigint I figured it out. I changed the event handler being used to set the follow type from `onChange` to `onMouseUp` and that now only sets the follow type after the dragging action is completed.

**Comment by Seroxdesign** [comment]
@bigint we've made the requested changes, mind taking another look at this when you get a chance?

**Comment by nitegeist** [comment]
@bigint any chance you can review this? I made the changes you requested.

**Comment by bigint** [comment]
Screenshot 2023-04-23 at 10 05 20 AM

Its stopping in the middle

**Comment by bigint** [comment]
Also make sure to ake the dropdown small 🙇🏼

**Comment by bigint** [comment]
Screenshot 2023-04-23 at 10 06 33 AM

On smaller devices it still stays down

**Comment by saagu335** [comment]
https://user-images.githubusercontent.com/69431456/233820185-0d752ad5-7c78-4dfe-a813-68cde1b71ae3.mov

CLS is happening too

**Comment by bigint** [comment]
image

Make sure slider looks same like in design

File: apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+import { useEffect, useState } from 'react';
 import { useReferenceModuleStore } from 'src/store/reference-module';
-import { Tooltip } from 'ui';
+import { Button, Modal, Tooltip } from 'ui';
+
+enum RANGE_VALUES {
+  MY_FOLLOWS = '1',
+  MY_FOLLOWERS = '30',
+  FRIENDS_OF_FRIENDS = '66',
+  EVERYONE = '100'
```
Comment (saagu335) [inline apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx:16]: Can't we make it like 0, 25, 50, 100?

File: apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx [hunk 4]
```diff
@@ -31,24 +49,135 @@ const ReferenceSettings: FC = () => {
...
+    isMyFollows
+      ? MY_FOLLOWS.value
+      : isMyFollowers
+      ? MY_FOLLOWERS.value
+      : isFriendsOfFriends
+      ? FRIENDS_OF_FRIENDS.value
+      : isEveryone
+      ? EVERYONE.value
+      : EVERYONE.value
+  );
```
Comment (saagu335) [inline apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx:62]: Can we move this to a function with switch condition? the readability is very low

File: apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx [hunk 5]
```diff
@@ -31,24 +49,135 @@ const ReferenceSettings: FC = () => {
...
 
   interface ModuleProps {
     title: string;
+    description: string;
     icon: ReactNode;
-    onClick: () => void;
-    selected: boolean;
   }
```
Comment (saagu335) [inline apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx:68]: Move props before the component starts

`const Module: FC<ModuleProps> = ({ title, icon, onClick, selected }) => (`

File: apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx [hunk 6]
```diff
@@ -31,24 +49,135 @@ const ReferenceSettings: FC = () => {
...
-          <div className="text-brand">{icon}</div>
-          <div>{title}</div>
+  const getClosest = (arr: string[], val: string): RANGE_VALUES => {
+    return arr.reduce(function (prev, curr) {
+      return Math.abs(Number(curr) - Number(val)) < Math.abs(Number(prev) - Number(val)) ? curr : prev;
+    }) as RANGE_VALUES;
+  };
+
+  const handlePointerUp = (event: React.PointerEvent<HTMLInputElement>) => {
+    let closest: RANGE_VALUES = getClosest(Object.values(RANGE_VALUES), event.currentTarget.value);
```
Comment (saagu335) [inline apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx:77]: I guess const works here

File: apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx [hunk 7]
```diff
@@ -31,24 +49,135 @@ const ReferenceSettings: FC = () => {
...
-        <div className="flex items-center space-x-1.5">
-          <div className="text-brand">{icon}</div>
-          <div>{title}</div>
+  const getClosest = (arr: string[], val: string): RANGE_VALUES => {
+    return arr.reduce(function (prev, curr) {
+      return Math.abs(Number(curr) - Number(val)) < Math.abs(Number(prev) - Number(val)) ? curr : prev;
+    }) as RANGE_VALUES;
+  };
+
+  const handlePointerUp = (event: React.PointerEvent<HTMLInputElement>) => {
```
Comment (saagu335) [inline apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx:76]: Treeshake only `PointerEvent`

File: apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx [hunk 8]
```diff
@@ -31,24 +49,135 @@ const ReferenceSettings: FC = () => {
...
+        setDegreesOfSeparation(2);
+        break;
+      case EVERYONE.value:
+        setSelectedReferenceModule(ReferenceModules.FollowerOnlyReferenceModule);
+        setOnlyFollowers(false);
+        break;
+      default:
+        break;
+    }
+  }, [followValue]);
```
Comment (saagu335) [inline apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx:102]: I'd suggest a read https://react.dev/learn/you-might-not-need-an-effect

File: apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx [hunk 9]
```diff
@@ -31,24 +49,135 @@ const ReferenceSettings: FC = () => {
...
+      )}
+      <div>
+        <div className="text-brand-500 text-lg font-semibold">{title}</div>
+        <div>{description}</div>
       </div>
-    </Menu.Item>
+    </div>
+  );
+
+  const Slider: FC = () => (
```
Comment (saagu335) [inline apps/web/src/components/Composer/Actions/ReferenceSettings/index.tsx:131]: Move this to a separate global component inside `packages/ui` so we can make sure it as a standard

**Review by saagu335 — CHANGES_REQUESTED** [review]

**Comment by saagu335** [comment]
@bigint We are planning for redesign #539 where do you think this fits?

**Comment by bigint** [comment]
@saagu335 Yes we need to think carefully before merging this to prod, new redesign works start from 27th, I guess we can pause this and continue after the redesign done.

**Pull request closed without merging**